chain/network/src/peer/peer_actor.rs

```diff
@@ -286,7 +290,7 @@ impl PeerActor {
         };
         let my_node_info = PeerInfo {
             id: network_state.config.node_id(),
-            addr: network_state.config.node_addr.as_ref().map(|a| **a),
+            addr: Some(stream.local_addr), // update own local address based on required stream
```

Comment:
What was the address stored previously here and why do we need to change it?

Reply:
it fails the handshake test for localnet. 
Previously stores the listening address, which can be set as 0.0.0.0 and optiona, now will store the actual connection address which will be set as 127.0.0.1 for localnet and is mandatory

Reply:
Ok, I think I understand what's going on from some of your other comments. `network_state.config.node_addr` has the ip address on which the node listens for new connections. When that listener does accept a connection and create a new stream, it is more correct/direct to just take the address for the stream from the stream object rather than assume it will be identical to the one for the listener and indirectly get the address through the network config.

Can you remove or modify this comment? It is not clear to me what is being said (for example we are not "updating" anything here, it is the spawn logic and everything is being set for the first time).

Reply:
Yes, you understood it perfectly. Took me some time diving into the code to figure out the differences between both socket addresses as well. 
I removed the comment instead of updating the comment.